### PR TITLE
Exercise 3

### PR DESCRIPTION
**Pull request by nicoletibaldi**

### CODE REVIEW

File: exercise-3/styles.css [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  padding-top: 5rem;
+  position: absolute;
+  left: -200rem;
+}
+
+input:checked + label {
+  display: block;
+  transition: 0.25s;
+  left: 31rem;
+}
```
Comment (drwpow) [inline exercise-3/styles.css:34]: This was the most important part of the slider 👍

File: exercise-3/styles.css [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+input:after {
```
Comment (drwpow) [inline exercise-3/styles.css:36]: Unfortunately, `input` elements don’t get pseudo-elements. At least, not according to the spec. This doesn’t work in Firefox 😞. Fine for the exercise, but just FYI for later!

screen shot 2018-06-07 at 6 48 13 am

File: exercise-3/index.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width">
+  <title>Exercise 3</title>
+  <link rel="stylesheet" href="./styles.css">
+</head>
+
+<body>
+  <div class="wrapper">
+    <input type="radio" id="1" name="carousel" checked/>
+    <label for="1"><img src="http://www.placecage.com/200/300"></label>
```
Comment (drwpow) [inline exercise-3/index.html:14]: I like your markup! It’s clean and minimal!

File: exercise-3/styles.css [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+label {
+  padding-top: 5rem;
+  position: absolute;
+  left: -200rem;
+}
+
+input:checked + label {
+  display: block;
+  transition: 0.25s;
+  left: 31rem;
```
Comment (drwpow) [inline exercise-3/styles.css:33]: This gets the job done, but it’s always better to use `transform: translate()` rather than animating basically anything else (`left`, `top`, `margin`, etc). This article breaks it down pretty well: https://www.paulirish.com/2012/why-moving-elements-with-translate-is-better-than-posabs-topleft/

TL;DR you’ll get stutter and < 60FPS

File: exercise-3/styles.css [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+label {
+  padding-top: 5rem;
+  position: absolute;
+  left: -200rem;
+}
+
+input:checked + label {
+  display: block;
+  transition: 0.25s;
```
Comment (drwpow) [inline exercise-3/styles.css:32]: The default easing curve isn’t linear in most browsers, so this isn’t bad, but we can be more expressive here! This is where the “personality” comes in.

The [animation pages](https://material.io/design/motion/understanding-motion.html#usage) of Google’s Material design manifesto not only provide some good `cubic-bezier()` curves (they’re in [Fondant](https://github.com/manifoldco/fondant/blob/master/src/theme.js#L146) btw!), but some important questions to ask when selecting a timing/easing:

- How large is this element—should it feel weighty, or light as a feather?
- Will this element reappear later? Should it animate from far off, or just off-screen?
- How is it moving? Is physics applying to it (translation, scale)? Or is it a light-based animation (color, opacity)?

Answering these questions can help us better lock in a timing and easing.

File: exercise-3/styles.css (hunk 2 above)
Comment (nicoletibaldi) [reply exercise-3/styles.css:36]: :( womp womp. I'll take a look at how you did it!
